### DISCUSSION TITLE
Suggestion for GitHub Integration Using GitHub Apps

### DISCUSSION THREAD

**Discussion by mtahle** [apache/devlake#4537] [Ideas]
Hello all,

Thank you for your hard work on this project. I would like to suggest adding a new feature to integrate with GitHub using "GitHub Apps". Currently, the solution uses personal access tokens, which can pose security risks and limit interactions. GitHub Apps offer better authentication methods and more flexibility for managing permissions. This change will enhance the integration's functionality and security.

Please let me know your thoughts on this suggestion.

Mujahed

**Comment by klesh** [comment]
Hi, @mtahle , thanks for the suggestion, we actually looked into the possibility before, it was hard for us to support it since we don't have the Github Apps and it didn't seem to be very popular at that time IIRC.

Would you mind posting a Feature Request on the github issues? Let's see how many people need this and we will support it if it got enough attention.

**Reply by mtahle** [reply]
Thanks for the response @klesh. I will submit a feature request soon
